kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,897 @@
+        givenNetworkServiceMockedForUsbInterfaceWithName("eth0");
+
+        whenGetInterfaceStatus("eth0", networkService);
+
+        thenNoExceptionIsThrown();
+        thenInterfaceStatusIsNotNull();
+        thenNetInterfaceTypeIs(NetworkInterfaceType.ETHERNET);
+    }
+
+    public void givenBasicMockedDbusConnector() throws DBusException, IOException {
+		when(dbusConnection.getRemoteObject(eq("org.freedesktop.NetworkManager"), eq("/org/freedesktop/NetworkManager"), any()))
+					.thenReturn(mockedNetworkManager);
+		
+		this.instanceNMDbusConnector = NMDbusConnector.getInstance(this.dbusConnection);
+	}
+
+    public void givenMockedPermissions() {
+
+        Map<String, String> tempPerms = new HashMap<>();
+
+        tempPerms.put("test1", "testVal1");
+        tempPerms.put("test2", "testVal2");
+        tempPerms.put("test3", "testVal3");
+
+        when(mockedNetworkManager.GetPermissions()).thenReturn(tempPerms);
+
+    }
+
+    public void givenMockedVersion() throws DBusException, IOException {
+
+        Properties mockProps = mock(org.freedesktop.dbus.interfaces.Properties.class);
+        when(mockProps.Get(eq("org.freedesktop.NetworkManager"), eq("Version"))).thenReturn("Mock-Version");
+
+        doReturn(mockProps).when(this.dbusConnection).getRemoteObject(eq("org.freedesktop.NetworkManager"),
+                eq("/org/freedesktop/NetworkManager"), eq(Properties.class));
+    }
+
+    public void givenMockedDevice(String interfaceName, NMDeviceType type, NMDeviceState state,
+            Boolean hasAssociatedConnection) throws DBusException, IOException {
+        Device mockedDevice1 = mock(Device.class);
+
+        this.mockDevices.put(interfaceName, mockedDevice1);
+
+        when(mockedDevice1.getObjectPath()).thenReturn("/mock/device/" + interfaceName);
+
+        Generic mockedDevice1Generic = mock(Generic.class);
+        when(mockedDevice1Generic.getObjectPath()).thenReturn("/mock/device/" + interfaceName);
+
+        DBusPath mockedPath1 = mock(DBusPath.class);
+        when(mockedPath1.getPath()).thenReturn("/mock/device/" + interfaceName);
+
+        Map<String, Map<String, Variant<?>>> mockedDevice1ConnectionSetting = new HashMap<String, Map<String, Variant<?>>>();
+        mockedDevice1ConnectionSetting.put("connection",
+                Collections.singletonMap("uuid", new Variant<>("mock-uuid-123")));
+
+        Settings mockedDevice1Settings = mock(Settings.class);
+        when(mockedDevice1Settings.GetConnectionByUuid(eq("mock-uuid-123"))).thenReturn(mockedPath1);
+
+        GetAppliedConnectionTuple mockedDevice1ConnectionTouple = mock(GetAppliedConnectionTuple.class);
+        when(mockedDevice1ConnectionTouple.getConnection()).thenReturn(mockedDevice1ConnectionSetting);
+
+        when(mockedDevice1.GetAppliedConnection(eq(new UInt32(0)))).thenReturn(mockedDevice1ConnectionTouple);
+
+        Properties mockedProperties1 = mock(Properties.class);
+
+        if (interfaceName.equals("lo")) {
```

Comment:
Additionally... why do you rely on the name of the device (which is set by the user during tests) when you can use the `NMDeviceType`?

If I need to update these tests I would expect the simulated properties to be built from the type and not the name of the interface.

Reply:
I changed this across the board, except for LO.
because it requires it to be type generic, so I used both properties. tell me what you think 
```
        if (type == NMDeviceType.NM_DEVICE_TYPE_GENERIC && interfaceName.equals("lo")) {
            when(mockedProperties1.Get(any(), any())).thenReturn("loopback");
        }

        if (type == NMDeviceType.NM_DEVICE_TYPE_WIFI) {
            simulateIwCommandOutputs(interfaceName, mockedProperties1);
        }
```